src/neuromaps_prime/graph.py

```diff
@@ -0,0 +1,603 @@
+
+The graph structure includes nodes representing brain template spaces
+and edges representing transformations between these spaces.
+
+nodes:
+    - name: Name of the brain template space (e.g., 'fsaverage', 'MEBRAINS')
+    - species: Species associated with the brain template space (e.g., 'human')
+    - description: Description of the brain template space
+    - surfaces: List of surface atlas resources available in this space
+    - volumes: List of volume atlas resources available in this space
+
+edges:
+    - surface_to_surface: Transformations between surface spaces
+    - volume_to_volume: Transformations between volume spaces
+
+see examples/example_graph_init.py for usage.
+
+## Make this more detailed later. ##
+
+"""
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Generator
+
+import networkx as nx
+import yaml
+
+from neuromaps_prime.models import (
+    SurfaceAtlasModel,
+    SurfaceTransformModel,
+    VolumeAtlasModel,
+    VolumeTransformModel,
+)
+
+
+@dataclass
+class Resource(ABC):
+    """Abstract base class for brain template resources."""
+
+    name: str
+    description: str
+    file_path: Path
+
+    @abstractmethod
+    def fetch(self) -> Path:
+        """Fetch the resource."""
+        pass
+
+
+@dataclass
+class SurfaceAtlas(Resource):
+    """Surface atlas for brain template spaces."""
+
+    space: str
+    density: str
+    hemisphere: str
+    resource_type: str
+
+    def fetch(self) -> Path:
+        """Fetch the surface resource."""
+        return self.file_path
+
+    def __repr__(self) -> str:
+        """Custom string representation for debugging."""
+        return f"{self.name}"
+
+
+@dataclass
+class SurfaceTransform(Resource):
+    """Transform resource between brain template spaces."""
+
+    source_space: str
+    target_space: str
+    density: str
+    hemisphere: str
+    resource_type: str
+
+    def fetch(self) -> Path:
+        """Fetch the transform resource."""
+        return self.file_path
+
+    def __repr__(self) -> str:
+        """Custom string representation for debugging."""
+        return self.name
+
+
+@dataclass
+class VolumeAtlas(Resource):
+    """Volume atlas for brain template spaces."""
+
+    space: str
+    resolution: str
+    resource_type: str
+
+    def fetch(self) -> Path:
+        """Fetch the volume resource."""
+        return self.file_path
+
+    def __repr__(self) -> str:
+        """Custom string representation for debugging."""
+        return self.name
+
+
+@dataclass
+class VolumeTransform(Resource):
+    """Transform resource between brain template spaces."""
+
+    source_space: str
+    target_space: str
+    resolution: str
+    resource_type: str
+
+    def fetch(self) -> Path:
+        """Fetch the transform resource."""
+        return self.file_path
+
+    def __repr__(self) -> str:
+        """Custom string representation for debugging."""
+        return self.name
```

Comment:
Would move some of these methods into the pydantic models and remove these dataclasses - having both introduces redundancy.